src/shared/util/logic-utils/logic-values-checker.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  IClientConditionSchema,
+  IField,
+  LogicConditionState,
+} from '../../../types'
+import {
+  isCheckboxField,
+  isDecimalField,
+  isDropdownField,
+  isRadioButtonField,
+  isRatingField,
+} from '../../../types/field/utils/guards'
+
+import {
+  ClientCheckboxCondition,
+  isClientCheckboxConditionValue,
+} from './checkbox'
+
+/**
+ * Checks if any of the values specified in logic are invalid.
+ * @param values Condition values.
+ * @param field Field with corresponding condition values.
+ * @param state Condition state
+ * @returns If the values are invalid
+ */
+export const checkIfHasInvalidValues = (
```

Comment:
don't really understand why this is in `shared`, since the implementation looks specific to the client-side data representation. would it make more sense to move this (and all the associated types and helpers) into a client-side service?